src/dc_job.rs

```diff
@@ -42,7 +42,7 @@ pub struct dc_job_t {
     pub desired_timestamp: i64,
     pub added_timestamp: i64,
     pub tries: libc::c_int,
-    pub param: *mut dc_param_t,
+    pub param: Option<Params>,
```

Comment:
Why optional?  You can have an empty params list.  That seems easier to handle.
(I recall some nice talk titled "the tyranny of None" or something, about ruby but pretty general applicable.  tl;dr None is just a null pointer with slightly better semantics, but avoiding them completely often reduces the amount of things to think about in random code)

Reply:
I don't have a strong preference in either direction